Add demo notebook showcasing GWLF-E API

rajadain opened the pull request:
## Overview

Adds a notebook doing a basic demo of the new API.

Connects #3525 

### Demo

https://github.com/WikiWatershed/model-my-watershed/blob/tt/add-api-demo/doc/MMW%20API%20GWLF-E%20Simple%20Tutorial.ipynb

## Testing Instructions

* Read through the notebook
  - [x] Ensure it makes sense

emilyhu0106 reviewed (approved):
The demo notebook is very well-written and touches all the parts of the new GWLF-E APIs. 

Wonder if it is possible to collapse the `Out [7]` cell that shows the full result from the `prepare` endpoint, or add a bookmark to jump to the next step? This cell takes up 3/4 of the lines and breaks the reading flow since it's in the middle of the demo. Clicking the blue bar at the left to collapse doesn't work for me. Maybe it is related to github's rendering?

rajadain commented:
Thanks for reviewing! I added `pprint(output, depth=3)` which limits the output somewhat and makes this more manageable. There are Jupyter Notebook renderers out there that support scrollable sections, but GitHub's does not do that.

I renamed the file to match the convention of existing ones. Will merge once green. Formatting issues can be revisited at a later date.
